### PR TITLE
[15.0][MIG] stock_return_request: Migration to 15.0

### PR DESCRIPTION
**Pull request by cesar-tecnativa**
@Tecnativa TT36575

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@cesar-tecnativa please check tests

**Comment by rousseldenis** [comment]
@cesar-tecnativa

**Comment by sergio-teruel** [comment]
tests fixed!!

File: stock_return_request/models/stock_return_request.py [hunk 1]
```diff
@@ -267,7 +264,7 @@ def _action_confirm(self):
                     return_move.product_uom_qty += qty
                 done_moves.setdefault(move, self.env["stock.move"])
                 done_moves[move] += return_move
-                return_move._action_confirm()
+                return_move.state = "confirmed"
```
Comment (rousseldenis) [inline stock_return_request/models/stock_return_request.py]: Why _action_confirm() should not be used anymore ? At least a comment should be set

File: stock_return_request/models/stock_return_request.py [hunk 2]
```diff
@@ -334,8 +331,9 @@ def _action_confirm(self):
             raise ValidationError(
                 _(
                     "It wasn't possible to assign stock for this returns:\n"
-                    "%s" % failed_moves_str
+                    "%(failed_moves_str)s"
                 )
+                % {"failed_moves_str": failed_moves_str}
```
Comment (rousseldenis) [inline stock_return_request/models/stock_return_request.py]: Usage of format() is better

File: stock_return_request/models/stock_return_request.py [hunk 3]
```diff
@@ -535,15 +533,17 @@ def _get_returnable_move_ids(self):
...
-                            line.product_uom_id.name,
-                            line.product_id.display_name,
-                            qty_found,
-                        )
+                        "It wasn't possible to find enough moves to return "
+                        "%(line_quantity)f %(line_product_uom_id_name)s"
+                        "of %(line_product_id_displayname)s. A maximum of %(qty_found)f "
+                        "can be returned."
                     )
+                    % {
```
Comment (rousseldenis) [inline stock_return_request/models/stock_return_request.py]: Same as above and for all strings formatting

**Review by rousseldenis — CHANGES_REQUESTED** [review]

File: stock_return_request/models/stock_return_request.py (hunk 2 above)
Comment (pedrobaeza) [reply stock_return_request/models/stock_return_request.py]: Even for translatable content?

Comment (rousseldenis) [reply stock_return_request/models/stock_return_request.py]: Of course, you can do:

`_("My string {variable1}, {variable2}").format(variable1="Test", variable2="Test2")`

Comment (pedrobaeza) [reply stock_return_request/models/stock_return_request.py]: Is that preference expressed in any place of the guidelines?

Comment (rousseldenis) [reply stock_return_request/models/stock_return_request.py]: It's not an OCA one, but a python one

https://realpython.com/python-f-strings/

Comment (rousseldenis) [reply stock_return_request/models/stock_return_request.py]: But we can do a PR on OCA contributing guidelines

Comment (pedrobaeza) [reply stock_return_request/models/stock_return_request.py]: f-strings is something different than `.format`. What are the advantages of using `format` over `%`?

Comment (rousseldenis) [reply stock_return_request/models/stock_return_request.py]: For me, readability

Comment (pedrobaeza) [reply stock_return_request/models/stock_return_request.py]: Using dictionaries makes the readability fairly the same, and with `%`, you have less chances of overpassing line length, so not sure it's an advantage.

**Comment by sergio-teruel** [comment]
All changes done!!

**Review by rousseldenis — APPROVED** [review]
LGTM

**Review by pedrobaeza — APPROVED** [review]

**Comment by rousseldenis** [comment]
@cesar-tecnativa @pedrobaeza Could you do a fix to remove api.Environment.manage() in hooks.py to avoid warnings in test logs ? Thanks

**Comment by pedrobaeza** [comment]
@rousseldenis done on #1134

File: stock_return_request/report/stock_return_report.xml [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+    <template id="report_stock_return_request">
+        <t t-foreach="docs" t-as="request">
+            <t t-call="stock_return_request.report_stock_return_request_document" />
+        </t>
+    </template>
+
+    <record id="action_report_stock_return_request" model="ir.actions.report">
+        <field name="name">Stock Return Request Report</field>
+        <field name="model">report.stock.card.report</field>
```
Comment (CasVissers-360ERP) [inline stock_return_request/report/stock_return_report.xml:131]: There is a dependency here on report.stock.card.report but this is not in the depends?

**Comment by CasVissers-360ERP** [comment]
@cesar-tecnativa I was testing this module but it seems it is missing dependencies that are blocking report generation?

**Pull request merged**